src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -46,7 +46,7 @@ public AssemblyName(string assemblyName)
             }
 
             if (parts._cultureName != null)
-                _cultureInfo = new CultureInfo(parts._cultureName);
+                _cultureInfo = CultureInfo.GetCultureInfo(parts._cultureName);
```

Comment:
`CultureInfo.GetCultureInfo` caches the CultureInfo instances. Avoids allocation and initialization costs.

Reply:
Will this cause any problems if someone tries to mutate AssemblyName.CultureInfo?

Reply:
I have undone this optimization from this PR. It can be done separately.